lib/Backend/Commands.hs

```diff
@@ -241,25 +241,45 @@ findCmd (FindOptions pathMb textMb sortMb filters filterFields) = do
       OpLTE -> (<=)
       OpEQ -> (==)
 
-renameCmd :: RenameOptions -> Sem r RenameResult
-renameCmd (RenameOptions _oldPath _newPath _force) = do
-  -- TODO: reuse copy+delete operations
-  undefined
+renameCmd
+  :: forall r.
+     ( Member BackendEffect r
+     , Member (Embed IO) r
+     , Member (Error CofferError) r
+     , Member (Error RenameResult) r
+     )
+  => RenameOptions -> Sem r RenameResult
+renameCmd (RenameOptions oldPath newPath force) = do
+  operations <- buildCopyOperations oldPath newPath force
+  void $ runCopyOperations operations
```

Comment:
Hmm it bothers me that we're using `void` to _swallow_ the result of `runCopyOperations` here...

* The type of `runCopyOperations` says it returns a `CopyResult` - which may be a successful `CopyResult` or a failed `CopyResult`. And if it's a failure, we shouldn't ignore it with `void`.
* However, in practice, the implementation of `runCopyOperations` always returns a successful `CopyResult`.

The crux of the problem seems to be that `runCopyOperations'`s type is misleading. It says it may fail or succeed, but that's not really true. Since it never returns a failed `CopyResult`, perhaps we should change its type to return a `Sem r ()` instead.

Reply:
Done